src/graphql/data-sources/cacheable-data-source.ts

```diff
@@ -81,13 +88,24 @@ export abstract class CacheableDataSource extends RESTDataSource {
     value: Value
   }): Promise<Entry<Value>> {
     const newEntry = { value, lastModified: this.environment.timer.now() }
-
     await this.environment.cache.set(key, newEntry)
-
     return newEntry
   }
 
-  public abstract updateCache(key: string): Promise<void>
+  private async lock(key: string): Promise<boolean> {
+    const lock = await this.environment.cache.setAndReturnPreviousValue<
+      boolean
+    >(this.getLockKey(key), true)
```

Comment:
```suggestion
    >(this.getLockKey(key), random-value)
```

When we add a `ttl` we should store a random value here which we use to unlock the key. See https://redis.io/topics/distlock#correct-implementation-with-a-single-instance for explanation.

Reply:
I don't really see how a random value is any better? I don't see a branch where the value is true and I didn't lock it myself (except some weird race condition but then we would execute both calls anyways and I'm fine with releasing the lock after either one was successful).

Reply:
From the linked document: "This is important in order to avoid removing a lock that was created by another client. For example a client may acquire the lock, get blocked in some operation for longer than the lock validity time (the time at which the key will expire), and later remove the lock, that was already acquired by some other client. Using just DEL is not safe as a client may remove the lock of another client. With the above script instead every lock is “signed” with a random string, so the lock will be removed only if it is still the one that was set by the client trying to remove it." -> we can also decide that this such a rare condition that we do not want to handle it...